### PR TITLE
exec credential provider: report original exec error message

### PR DESCRIPTION
**Pull request by stefreak**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Execution of a command can fail for a variety of reasons; The most common reason indeed is that the binary wasn't found in the PATH, but there is a plethora of other possible reasons.

This commit makes sure not to swallow the original error message, which might be misleading if there is another underlying reason for the error.

#### Which issue(s) this PR fixes:
I did not create an issue for this

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by stefreak** [comment]
Hello @yliaog @enj @mikedanese Could one of you have a quick look and `/lgtm` the PR? According tothe contributor guide that's the next step, but I didn't contribute very often and don't know exactly what to do here. https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md#the-code-review-process

Thank you for your time ❤️

**Comment by stefreak** [comment]
@fedebongio thank you for the initial triage. The assigned owners/ reviewers have not reacted since then, what can I do to help?

**Comment by liggitt** [comment]
test failure is relevant:

```
    exec_test.go:512: got "exec: exec: \"does not exist\": executable file not found in $PATH\n\nIt looks like you are trying to use a client-go credential plugin that is not installed.\n\nTo learn more about this feature, consult the documentation available at:\n      https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins", wanted "exec: executable does not exist not found..."
    --- FAIL: TestExecPluginViaClient/unknown_binary (0.00s)
;=== RUN   TestExecPluginViaClient/unknown_binary_v1beta1
    exec_test.go:512: got "exec: exec: \"does not exist\": executable file not found in $PATH\n\nIt looks like you are trying to use a client-go credential plugin that is not installed.\n\nTo learn more about this feature, consult the documentation available at:\n      https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins", wanted "exec: executable does not exist not found..."
    --- FAIL: TestExecPluginViaClient/unknown_binary_v1beta1 (0.00s)
```

it looks like we probably should tweak the format to avoid the double `exec: exec:` prefix

File: staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go [hunk 1]
```diff
@@ -520,9 +520,12 @@ func (a *Authenticator) refreshCredsLocked() error {
...
 func (a *Authenticator) wrapCmdRunErrorLocked(err error) error {
 	switch err.(type) {
-	case *exec.Error: // Binary does not exist (see exec.Error).
+	case *exec.Error: // Failed to execute binary (see exec.Error).
 		builder := strings.Builder{}
-		fmt.Fprintf(&builder, "exec: executable %s not found", a.cmd)
+		// The most common reason for failing to execute a binary is that it wasn't found.
+		// There are more failure modes too though, and we need to preserve the original
+		// error message to avoid swallowing important clues.
+		fmt.Fprintf(&builder, "exec: %v", err)
```
Comment (ardaguclu) [inline staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go:528]: What about;
```suggestion
		fmt.Fprintf(&builder, "executable %s error out: %w", a.cmd, err)
```

**Review by ardaguclu — COMMENTED** [review]
I think, this indeed makes troubleshooting easier.

**Comment by stefreak** [comment]
Thank your time, I will have a look at the test failures & push a fix.

**Pull request closed without merging**